app/models/grda_warehouse/tasks/push_clients_to_cas.rb

```diff
@@ -72,7 +74,7 @@ def sync!
             GrdaWarehouse::Hud::Project.find_each do |project|
               safe_project_names[project.id] = project.safe_project_name
             end
-            client_source.preload(preloads).
+            client_source.lazy_preload(preloads).
```

Comment:
Should we check if this is still needed after rails upgrade?
```suggestion
            TodoOrDie('test behavior after rails upgrade', if: Rails.version !~ /\A7\.0/)
            client_source.lazy_preload(preloads).
```

Reply:
Sure, let's revisit.